giskard_vision/landmark_detection/detectors/base.py

```diff
@@ -0,0 +1,65 @@
+from abc import abstractmethod
+from typing import Any, Sequence
+
+from giskard_vision.detectors.base import DetectorVisionBase, ScanResult
+from giskard_vision.landmark_detection.tests.base import TestDiff
+from giskard_vision.landmark_detection.tests.performance import NMEMean
+
+
+class LandmarkDetectionBaseDetector(DetectorVisionBase):
+    """
+    Abstract class for Landmark Detection Detectors
+
+    Methods:
+        get_dataloaders(dataset: Any) -> Sequence[Any]:
+            Abstract method that returns a list of dataloaders corresponding to
+            slices or transformations
+
+        get_results(model: Any, dataset: Any) -> Sequence[ScanResult]:
+            Returns a list of ScanResult containing the evaluation results
+
+        get_scan_result(self, test_result) -> ScanResult:
+            Convert TestResult to ScanResult
+    """
+
+    @abstractmethod
+    def get_dataloaders(self, dataset: Any) -> Sequence[Any]:
+        ...
+
+    def get_results(self, model: Any, dataset: Any) -> Sequence[ScanResult]:
+        dataloaders = self.get_dataloaders(dataset)
+
+        results = []
+        for dl in dataloaders:
+            test_result = TestDiff(metric=NMEMean, threshold=1).run(
+                model=model,
+                dataloader=dl,
+                dataloader_ref=dataset,
+            )
+            results.append(self.get_scan_result(test_result))
+
+        return results
+
+    def get_scan_result(self, test_result) -> ScanResult:
+        try:
+            from giskard.scanner.issues import IssueLevel
+        except (ImportError, ModuleNotFoundError) as e:
+            e.msg = "Please install giskard to use custom detectors"
+            raise e
```

Comment:
let's use this instead: https://github.com/Giskard-AI/giskard-vision/blob/17b969fc062ee3b11173a59c3868e01ec6a64238/giskard_vision/landmark_detection/dataloaders/loaders.py#L229